src/crypto/abe/hybrid_crypto/encrypted_header.ts

```diff
@@ -0,0 +1,84 @@
+import { logger } from "../../../utils/logger"
+import { deserializeList, toBeBytes } from "../../../utils/utils"
+import { SYMMETRIC_KEY_SIZE } from "./encryption_parameters"
+
+export class EncryptedHeader {
+  private _symmetricKey: Uint8Array
+  private _encryptedSymmetricKey: Uint8Array
+  private _encryptedSymmetricKeySizeAsArray: Uint8Array
+
+  public get encryptedSymmetricKey(): Uint8Array {
+    return this._encryptedSymmetricKey
+  }
+  public set encryptedSymmetricKey(value: Uint8Array) {
+    this._encryptedSymmetricKey = value
+  }
+
+  public get symmetricKey(): Uint8Array {
+    return this._symmetricKey
+  }
+  public set symmetricKey(value: Uint8Array) {
+    this._symmetricKey = value
+  }
+
+  public get encryptedSymmetricKeySizeAsArray(): Uint8Array {
+    return this._encryptedSymmetricKeySizeAsArray
+  }
+  public set encryptedSymmetricKeySizeAsArray(value: Uint8Array) {
+    this._encryptedSymmetricKeySizeAsArray = value
+  }
+
+  constructor(symmetricKey: Uint8Array, encryptedSymmetricKey: Uint8Array) {
+    this._symmetricKey = symmetricKey
+    this._encryptedSymmetricKey = encryptedSymmetricKey
+
+    // Convert symmetric key length to 4-bytes array
+    this._encryptedSymmetricKeySizeAsArray = toBeBytes(this._encryptedSymmetricKey.length)
+  }
+
+  /**
+   * Deserialize an ABE encrypted header
+   *
+   * @param headerBytes an ABE header
+   * @returns the corresponding EncryptedHeader object
+   */
+  public static parse(headerBytes: Uint8Array): EncryptedHeader {
```

Comment:
Since it's parsing JSON maybe accept `string` instead of bytes?

Reply:
I must say you are right but function became deprecated. Removing it.